api/config.js

```diff
@@ -0,0 +1,63 @@
+import {createRemoteJWKSet, jwtVerify} from "jose";
+import request from "needle";
+
+const {AUTH_SERVER_URL, AUTH_REALM, GATEWAY_ACCOUNT_URL, GATEWAY_KEY} = process.env;
+// Credit: https://github.com/keycloak/keycloak-nodejs-connect/issues/492#issuecomment-1603213677
+const jwks = createRemoteJWKSet(new URL(
+	`${AUTH_SERVER_URL}/realms/${AUTH_REALM}/protocol/openid-connect/certs`));
+const GATEWAY = new class {
+	accountUrl = new URL(GATEWAY_ACCOUNT_URL);
+	accountName = this.accountUrl.pathname.slice(1);
+};
+
+// noinspection JSUnusedGlobalSymbols
+/**
+ * @param {import('@vercel/node').VercelRequest} req
+ * @param {import('@vercel/node').VercelResponse} res
+ */
+export default async function (req, res) {
+	const sendErr = (code, text, error) => {
+		console.warn(error);
+		res.status(code).json({text, error});
+	};
+	const throwErr = (code, text) => error => {
+		throw {send: () => sendErr(code, text, error)};
+	};
+	try {
+		const subdomain = req.query.domain;
+		const key = req.body;
+		const [type, token] = req.headers.authorization?.split(" ") ?? [];
+		if (type !== "Bearer")
+			return sendErr(401, 'Unauthorised');
+		const {payload} = await jwtVerify(token, jwks)
+			.catch(throwErr(401, 'Unauthorised'));
```

Comment:
The Identity Provider token is provided as a `Bearer` token in the `Authorization` HTTP header, and validated using the Identity Provider JWK keyset. Note that this is performing local JWT verification and not normally making a network call to the Identity Provider; but the `jose` library will periodically update the keyset from the provider.